### PR TITLE
close modal if hotkey is pressed

### PR DESCRIPTION
**Pull request by LuqDragon**
hotkey

### CODE REVIEW

**Comment by RedSparr0w** [comment]
I think we might have some functions bound to the `close` button on the gym leader modals.
So we might also need to trigger those if the modal is/was visible.

**Comment by RedSparr0w** [comment]
Or instead of triggering the close function, we could just ignore this input if any modals are currently visible?

**Comment by LuqDragon** [comment]
dont have a close function, just `data-dismiss="modal"` that make the same thing then JQuery right?

the idea is not close the modal?

**Comment by RedSparr0w** [comment]
Ah, I misremembered, it looks like the function is run when we show the modal, not on close.

**Comment by LuqDragon** [comment]
want to implement this as a function to all modals? checking if they a showing?

**Comment by RedSparr0w** [comment]
Currently getting this error:
image

May be easier just to check if any modals are visible, as we likely wouldn't want to allow the battle to restart when covered.

**Comment by LuqDragon** [comment]
i tested three methods:
$('#myModal').is(':visible');   ->  dont work with auto clicker
$('#myModal').data('bs.modal').isShown ;    -> work but we get the error
$('#myModal').hasClass('in');    ->  never works

maybe join them?

**Comment by RedSparr0w** [comment]
> i tested three methods:
> `$('#myModal').is(':visible');` -> dont work with auto clicker
> `$('#myModal').data('bs.modal').isShown;` -> work but we get the error
> `$('#myModal').hasClass('in');` -> never works
> 
> maybe join them?

These ones might also work (haven't tried them out yet)
```js
$('#receiveBadgeModal').hasClass('show')
// or just for any modal
$('.modal.show').length
$('.modal').hasClass('show')
```

**Comment by LuqDragon** [comment]
the problem with `hasClass` is that you spam the number or keep the button pressed, the modal get the `show` class after the modal is fully opened, change to another champion before the modal

File: src/scripts/GameController.ts [hunk 1]
```diff
@@ -149,10 +149,16 @@ class GameController {
                     }
                     e.preventDefault();
                 } else if ('gymList' in player.town()) {
-                    const number = Number(e.key);
-                    // Check if a number higher than 0 and less than total Gyms was pressed
-                    if (number && number <= player.town().gymList().length) {
-                        GymRunner.startGym(player.town().gymList()[number - 1]());
+                    // Close receiveBadgeModal before start another champion
+                    //$('#receiveBadgeModal').removeClass('fade').modal('hide').addClass('fade');
+
```
Comment (RedSparr0w) [inline src/scripts/GameController.ts:154]:
```suggestion
```

**Review by RedSparr0w — APPROVED** [review]
Changes working great for me!

Thanks for the PR 👍

**Pull request merged**